src/packet_types.c

```diff
@@ -79,3 +79,39 @@ void signal_report_init(SignalReportBlock *b, const int8_t snr, const int8_t rss
     b->contents.request = request;
     b->contents._dead_space = 0;
 }
+/**
+ * Initializes an altitude data block with the provided information.
+ * @param b The altitude data to be initialized
+ * @param measurment_time The mission time at the taking of the measurment
+ * @param pressure The measured pressure in terms of Pascals. This field is a signed 32 bit integer in two's complement
+ * format.
+ * @param temperature The measured temperature in units of 1 millidegree Celsius/LSB. This field is a signed 32 bit
+ * integer in two's complement format.
+ * @param altitude The calculated altitude in units of 1 mm/LSB. This field is a signed 32 bit integer in two’s
+ * complement format.
+ */
+void altidude_data_block_init(AltitudeDataBlock *b, const int8_t measurment_time, const uint8_t pressure,
+                              const uint8_t temperature, const uint8_t altitude) {
+    b->contents.measurment_time = measurment_time;
```

Comment:
You've implemented the altitude data block and angular velocity data block as byte arrays, which is good because that's what we're going to finalize our design with. However, arrays don't have named members, so you can't access measurement time using `b->contents.measurement_time`.

In your array, measurement time is the first 4 bytes (or 32 bits). You'll have to initialize all your fields more like this:

```c
// Measurement time
memcpy(b->bytes, &measurement_time, sizeof(uint32_t));
```